examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+      "    input_node = nengo.Node(output=input_fun, label='input')\n",
+      "    nengo.Connection(input_node, assoc_mem.input)\n",
+      "\n",
+      "    input_probe = nengo.Probe(input_node)\n",
+      "    output_probe = nengo.Probe(assoc_mem.output, synapse=0.03)\n",
+      "    \n",
+      "sim = nengo.Simulator(model4)\n",
+      "sim.run(1.)\n",
+      "t = sim.trange()\n",
+      "\n",
+      "input_net = sim.data[input_probe]\n",
+      "output_net = sim.data[output_probe]\n",
+      "\n",
+      "plot_similarities(input_net, output_net, vocab_numbers, top=1.1)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "So far we have created a model which realises addition by one. The next step is to automatise this, so that when the model sees `ONE` it will produce `TWO, THREE, FOUR, FIVE`.\n",
+      "To achieve counting, we need to introduce a feedback connection. That is, the network output needs to be fed into its input at the next time step. This can be easily done in Nengo by adding just one additional connection. Now, we initialise the simulation by presenting the semantic pointer `ONE` at the input for the duration of 0.2 simulation time. "
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "def input_fun(t):\n",
+      "    if 0 < t < 0.2:\n",
+      "        return vocab_numbers['ONE'].v\n",
+      "    else:\n",
+      "        return np.zeros(dim)\n",
+      "\n",
+      "with nengo.Network('Counting', seed=3) as model5:\n",
+      "    assoc_mem = spa.AssociativeMemory(input_vocab=vocab_numbers, output_vocab=vocab_numbers,\n",
+      "                                      input_keys=input_keys, output_keys=output_keys, \n",
+      "                                      wta_output=True)\n",
+      "    \n",
+      "    input_node = nengo.Node(output=input_fun, label='input')\n",
+      "    nengo.Connection(input_node, assoc_mem.input)\n",
+      "\n",
+      "    # added feedback connection\n",
+      "    nengo.Connection(assoc_mem.output, assoc_mem.input, synapse=.18, transform=5)\n",
+      "    \n",
+      "    input_probe = nengo.Probe(assoc_mem.input)\n",
+      "    output_probe = nengo.Probe(assoc_mem.output, synapse=0.03)\n",
+      "    \n",
+      "sim = nengo.Simulator(model5)\n",
+      "sim.run(1.)\n",
+      "t = sim.trange()\n",
+      "\n",
+      "input_net = sim.data[input_probe]\n",
+      "output_net = sim.data[output_probe]\n",
+      "\n",
+      "plot_similarities(input_net, output_net, vocab_numbers)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "The only input our model receives is `ONE` at the beginning of the simulation. After that, it produces a sequence of consecutive numbers up to five."
```

Comment:
If you ask for similarity of the vector with output vectors, it's because we use synaptic delay in probes to simulate how this input will look like for another module that connects to associative memory. And the same holds then for the inputs in this examples, which are just rewired outputs.